lib/gooddata/bricks/middleware/execution_result_middleware.rb

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2019, GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+module GoodData
+  module Bricks
+    module ExecutionStatus
+      OK = 'OK'
+      ERROR = 'ERROR'
+      WARNING = 'WARNING'
+    end
+
+    class ExecutionResultMiddleware < Bricks::Middleware
+      def call(params)
+        params = self.class.update_execution_result(params, params["EXECUTION_RESULT_STATUS"], params["EXECUTION_RESULT_MESSAGE"])
+        @app.call(params)
+      end
+
+      def self.update_execution_result(params, status, message = "")
```

Comment:
add ruby doc: Update process execution result when the script is executed inside a GDC ETL process execution (ruby bricks). Ruby bricks should update execution result at the end of script with status=WARNING or OK and summary message if the script exits normally. If any fatal error, script should update execution result with status=ERROR and error message, then throw exception to notify GDC platform that the script exists unexpectedly.

Reply:
I updated the ruby doc.